src/agl/WebRuntimeAGL.cpp

```diff
@@ -79,36 +79,37 @@ class AGLRendererDelegateWAM : public webos::WebOSMainDelegate {
     }
 };
 
-void Launcher::register_surfpid(pid_t surf_pid)
+void Launcher::register_surfpid(pid_t app_pid, pid_t surf_pid)
 {
-  if (surf_pid == m_rid) {
-    if (!std::count(m_pid_v.begin(), m_pid_v.end(), surf_pid)) {
-      fprintf(stderr, "surface creator(pid=%d) registered\r\n", surf_pid);
-      m_pid_v.push_back(surf_pid);
-      fprintf(stderr, "m_pid_v.count(%d) = %d\r\n", surf_pid,
-                (int)std::count(m_pid_v.begin(), m_pid_v.end(), surf_pid));
-    }
+  if (app_pid != m_rid)
+    return;
+  auto surface_id = m_pid_map.find(app_pid);
+  if (surface_id != m_pid_map.end()) {
+    fprintf(stderr, "register_surfpid, (app_pid=%d) already registered surface_id with (surface_id=%d)\r\n",
+            (int)app_pid, (int)surface_id->second);
+    return;
   }
+
+  m_pid_map.emplace(app_pid, surf_pid);
```

Comment:
Shouldn't it be enough to run insert?

```
auto surface_id = m_pid_map.insert(app_pid, surf_pid);
if (surface_id != m_pid_map.end()) {
  ...
  return;
}
```

Reply:
Done. As I couldn't get it as 'auto', I updated it a little bit from your suggestion.